test/ref_ops_test.cpp

```diff
@@ -2674,6 +2674,165 @@ TEST_CASE(gathernd_test)
     }
 }
 
+TEST_CASE(gathernd_dynamic)
+{
+    { // dynamic data, all dimensions fixed
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+
+        migraphx::shape ds{migraphx::shape::float_type, {{2, 2, 2}, {3, 3, 0}, {1, 1, 0}}};
+        migraphx::shape is{migraphx::shape::int64_type, {2, 2, 1}};
+
+        std::vector<float> data_vec(2 * 3 * 1);
+        std::iota(data_vec.begin(), data_vec.end(), 0);
+        std::vector<int64_t> indices_vec{1, 0, 0, 1};
+
+        auto xdata  = mm->add_parameter("X", ds);
+        auto xindex = mm->add_parameter("I", is);
+
+        auto gathernd_op = migraphx::make_op("gathernd");
+        auto gathernd    = mm->add_instruction(gathernd_op, xdata, xindex);
+
+        mm->add_return({gathernd});
+        p.compile(migraphx::ref::target{});
+
+        migraphx::parameter_map params;
+        migraphx::shape input_fixed_shape0{migraphx::shape::float_type, {2, 3, 1}}; // data
+        migraphx::shape input_fixed_shape1{migraphx::shape::int64_type, {2, 2, 1}}; // index
+
+        params["X"] = migraphx::argument(input_fixed_shape0, data_vec.data());
+        params["I"] = migraphx::argument(input_fixed_shape1, indices_vec.data());
+
+        auto result = p.eval(params).back();
+        std::vector<float> res_data{};
+        std::vector<float> gold{3, 4, 5, 0, 1, 2, 0, 1, 2, 3, 4, 5};
+        result.visit([&](auto output) { res_data.assign(output.begin(), output.end()); });
+
+        EXPECT(migraphx::verify_range(res_data, gold));
+    }
```

Comment:
This case shouldn't happen because it can be easily simplified to a static shape.